app/containers/ItunesContainer/index.js

```diff
@@ -0,0 +1,199 @@
+import React, { useEffect, memo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import get from 'lodash/get';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+import { Card, Skeleton, Input } from 'antd';
+import styled from 'styled-components';
+import { injectIntl } from 'react-intl';
+import { useHistory } from 'react-router-dom';
+import T from '@components/T';
+import Clickable from '@components/Clickable';
+import { useInjectSaga } from 'utils/injectSaga';
+import { selectItunesContainer, selectSongsData, selectSongsError, selectSongName } from './selectors';
+import { itunesContainerCreators } from './reducer';
+import saga from './saga';
+
+const { Search } = Input;
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px 0;
+    max-width: ${props => props.maxwidth};
+    color: ${props => props.color};
+    ${props => props.color && `color: ${props.color}`};
+  }
+`;
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: ${props => props.maxwidth}px;
+    width: 100%;
+    margin: 0 auto;
+    padding: ${props => props.padding}px;
+  }
+`;
+const RightContent = styled.div`
+  display: flex;
+  align-self: flex-end;
+`;
+export function ItunesContainer({
+  dispatchItunesSongs,
+  dispatchClearItunesSongs,
+  intl,
+  songsData = {},
+  songsError = null,
+  songName,
+  maxwidth,
+  padding
+}) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(songsData, 'results', null) || songsError;
+    if (loading && loaded) {
+      setLoading(false);
+    }
+  }, [songsData]);
+
+  useEffect(() => {
+    if (songName && !songsData?.results?.length) {
+      dispatchItunesSongs(songName);
+      setLoading(true);
+    }
+  }, []);
+  useInjectSaga({ key: 'itunesContainer', saga });
+
+  const history = useHistory();
+
+  const handleOnChange = rName => {
+    if (!isEmpty(rName)) {
+      dispatchItunesSongs(rName);
+      setLoading(true);
+    } else {
+      dispatchClearItunesSongs();
+    }
+  };
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderRepoList = () => {
+    const items = get(songsData, 'results', []);
+    const resultCount = get(songsData, 'resultCount', 0);
+    return (
+      (items.length !== 0 || loading) && (
```

Comment:
We can use If component from `@components/If`

Reply:
Ok ill check it out